### PR TITLE
Make bloop more tolerant to network failures

### PR DESCRIPTION
**Pull request by rsdy**
There are 2 bugs fixed here:

1.  bloop wouldn't start without network connection
2.  we silence reporting Error conditions when we suspect it's stemming from the network (there's no exhaustive list of error states we know would be indicative of this, though)

1) requires the `credentials.json` migration, because now we persist the name of the user that's currently logged in (to validate if there's actually a logged in user). Originally the frontend blew up on start without network connection because `github/status` blew up because sentry blew up. So now all of this is channeled through `/config` which doens't blow up.

### CODE REVIEW

**Review by ggordonhall — CHANGES_REQUESTED** [review]
Dropping and restarting network connection while a repo is in state `Cloning` causes the repo to get stuck in the `Queued` state.

File: server/bleep/src/intelligence/language/python/mod.rs [hunk 1]
```diff
@@ -159,10 +159,6 @@ mod tests {
                     imports: [
                         List {
                             context: "from typings import §List§",
```
Comment (ggordonhall) [inline server/bleep/src/intelligence/language/python/mod.rs:161]: Why has this been removed?

File: server/bleep/src/indexes/file.rs [hunk 2]
```diff
@@ -532,7 +532,7 @@ impl RepoDir {
         relative_path: &Path,
         repo_disk_path: &Path,
         repo_ref: &str,
-        last_commit: u64,
+        last_commit: i64,
```
Comment (ggordonhall) [inline server/bleep/src/indexes/file.rs:535]: Why the sign change?

Comment (rsdy) [reply server/bleep/src/indexes/file.rs:535]: It's a dependency requirement, gix changed fixed their apis to track the correct type for unix seconds. (previously it was different from the underlying POSIX APIs for some reason)

File: server/bleep/src/intelligence/language/python/mod.rs (hunk 1 above)
Comment (rsdy) [reply server/bleep/src/intelligence/language/python/mod.rs:161]: This is an expect test, this change is a result of upgrading the upstream deps.

Comment (nerdypepper) [reply server/bleep/src/intelligence/language/python/mod.rs:161]: these tests shouldn't change on version upgrade. will take a look at what is going on here.

Comment (nerdypepper) [reply server/bleep/src/intelligence/language/python/mod.rs:161]: can we roll-back the change to python version dependency? it contains a breaking change to the grammar.

File: server/bleep/src/state.rs [hunk 3]
```diff
@@ -45,6 +40,10 @@ pub struct StateSource {
     #[clap(long)]
     #[serde(default)]
     cookie_key: Option<PathBuf>,
+
+    #[serde(skip)]
+    #[clap(skip)]
+    root_dir: PathBuf,
```
Comment (nerdypepper) [inline server/bleep/src/state.rs:46]: what is root-dir? would be nice to have a doc-comment here.

**Review by nerdypepper — APPROVED** [review]

**Review by ggordonhall — APPROVED** [review]

**Pull request merged**